File: search_ASL/scripts/03_retrieve_novel_hosts.R
# author: dlueckin
# date: Tue May 30 15:06:06 2023

# libraries ---------------------------------------------------------------
library(data.table)
library(stringr)
library(seqinr)
library(tidyr)
library(ggplot2)
library(dplyr)

# working directory -------------------------------------------------------
this_dir <- "/home/dlueckin/projects/pR1SE_australian_lakes/search_ASL/scripts"
setwd(this_dir)
print(paste0("Setting wd to: \n ", this_dir))


# read contig_df -----------------------------------------------------------

contig_df <- fread("novel_hosts_df.tsv")


# read all intput fasta files and keep relevant hits ----------------------
relevant_seqs <- list()

for(fasta_file in list.files("../tal/4_Dom/", pattern = "fasta", full.names = TRUE)){
    current_fasta <- read.fasta(fasta_file)
    relevant_seqs <- c(relevant_seqs, current_fasta[which(names(current_fasta) %in% contig_df$contig)])
}


# save to file ------------------------------------------------------------

write.fasta(relevant_seqs, names = names(relevant_seqs), file.out = "relevant_sequences.fasta")

File: search_ASL/scripts/03_retrieve_novel_hosts.py
from Bio import SeqIO
import pandas as pd
import os
import re

# Read contig_df
contig_df = pd.read_csv("novel_hosts_df.tsv")

# Read all input fasta files and keep relevant hits
relevant_seqs = []

fasta_files = [os.path.join("../tal/4_Dom/", file) for file in os.listdir("../tal/4_Dom/") if file.endswith(".fasta")]

for fasta_file in fasta_files:
    for record in SeqIO.parse(fasta_file, "fasta"):
        short_name = re.sub("\_length.*", "", record.name)
        if short_name in contig_df["contig"].values and record.name not in [record.name for record in relevant_seqs]:
            relevant_seqs.append(record)
            print(short_name)

# Save to file
SeqIO.write(relevant_seqs, "novel_host_sequences.fasta", "fasta")
